crypto/crypto/src/hash.rs

```diff
@@ -356,6 +356,30 @@ impl<'a> std::iter::DoubleEndedIterator for HashValueBitIterator<'a> {
 
 impl<'a> std::iter::ExactSizeIterator for HashValueBitIterator<'a> {}
 
+impl std::iter::FromIterator<bool> for HashValue {
+    fn from_iter<I: IntoIterator<Item = bool>>(iter: I) -> Self {
```

Comment:
seems to me a little bit more straightforward if you do

    let mut buf = [0;LENGTH_IN_BYTES];
    // iterate, set, and count
    // assert count
    HashValue::from_slice(&buf)